### ISSUE TITLE
Feature: Restore shared notes (from prev meeting)

### ISSUE THREAD

**Issue by ranjithrajv** [bigbluebutton/bigbluebutton#13432]
**Is your feature request related to a problem? Please describe.**
The agenda of the meeting might not complete in a single meeting, it may be needed to take breaks in-between or meet over regular intervals to complete the agenda. Downloading shared notes and struggling to re-arrange the markdown is a hectic process. 

**Describe the solution you'd like**
Please allow saving shared notes before closing the meeting and restore the shared notes from previous meeting when the room is re-opened. 

**Describe alternatives you've considered**
- Download shared notes at the end of every meeting 
- Copy-pasting the contents to shared notes after reopening the rooms

**Additional context**

**Comment by ranjithrajv** [comment]
Seems you might have misunderstood the requirement.

We are using the export option from long time. But the feature request I was asking is BBB room restoring shared notes (on opening) from previous meeting, without the need to export/import.

Please reopen the issue, as the asked feature is yet to be implemented.

**Comment by antobinary** [comment]
Hi @ranjithrajv ,
I am reopenning and tweaking the title. I do not really see how this can be implemented with the current architecture of the shared notes. I will still leave this open in case others want to comment.

**Comment by ranjithrajv** [comment]
I've noticed the BBB team is planning to make architectural changes to Shared notes, here https://github.com/bigbluebutton/bbb-pads. Does the new architecture allow restoring shared notes from previous sessions?

**Comment by pedrobmarin** [comment]
No.

The only way I see something like this feasible is adding an API call to give the shared notes some text. BigBlueButton is not prev/next meeting aware. You would need Greenlight or similar to know that a meeting is a continuation of another and then handle this shared notes content injection.

**Comment by MWWorks** [comment]
> The only way I see something like this feasible is adding an API call to give the shared notes some text.

This would be useful - I often have this issue of wanting to persist shared notes and would be happy to DIY it through the API. But unless I'm missing something, we'd also need an API call to be able to get the shared notes (eg on meeting end) - I don't think this exists either?

**Comment by danielpetri1** [comment]
@MWWorks Playback of recordings contain the shared notes, so you can get the raw HTML data from the recording's data. Alternatively there is a 'notes' format I believe that can be enabled to process the shared notes after the meeting ends.

The contents of this file could then be re-added with Etherpad's setHTML API call.

**Comment by ffdixon** [comment]
It's a good suggestion -- lets leave it open.

**Comment by andreas-urra** [comment]
I agree, great suggestion. Actually, I was looking for a way to import previously exported shared notes and couldn't find one. Interestingly, the export dialogue in BBB says "Import/Export". Did I miss something or is the import function available somewhere else?
My use case would be to have a set of templates for shared notes, e.g. for protocols, agendas etc.
Copy/Paste from previously exported notes doesn't stay completely true to the formatting. So it is a lot of manual work at the beginning of a meeting.

**Comment by kaystrobach** [comment]
Normally etherpad can do this, technically it seems to be disabled ...
